src/index.ts

```diff
@@ -485,3 +493,126 @@ export class CruxClient {
     }
 
 }
+
+// DDD Experimental Stuff
+export interface ICruxOnBoardingClientOptions {
+    cruxDomainRepository?: ICruxDomainRepository;
+    cruxAssetTranslatorRepository?: ICruxAssetTranslatorRepository;
+    configKeyManager?: IKeyManager;
+    cacheStorage?: storage.StorageService;
+}
+
+export class CruxOnBoardingClient {
+    private _initPromise: Promise<void>;
+    private _cruxDomainRepository: ICruxDomainRepository;
+    private _cruxAssetTranslatorRepository: ICruxAssetTranslatorRepository;
+    private _configKeyManager?: IKeyManager;
+    private _cruxDomain?: CruxDomain;
+
+    constructor(options: ICruxOnBoardingClientOptions) {
+        cacheStorage = options.cacheStorage || new inmemStorage.InMemStorage();
+        this._cruxDomainRepository = options.cruxDomainRepository || new BlockstackCruxDomainRepository({});
+        this._cruxAssetTranslatorRepository = options.cruxAssetTranslatorRepository || new BlockstackCruxAssetTranslatorRepository();
+        this._configKeyManager = options.configKeyManager;
+        // restore the wallet on-boarding client with the key provided
+        this._initPromise = this._init();
+        log.info("CruxOnBoardingClient initialised");
+    }
+
+    public isCruxDomainAvailable = async (domain: string): Promise<boolean> => {
+        return this._cruxDomainRepository.find(domain);
+    }
+
+    // todo
+    public registerCruxDomain = async (domain: string): Promise<void> => {
+        if (!this._configKeyManager) {
+            throw errors.ErrorHelper.getPackageError(null, errors.PackageErrorCode.PrivateKeyRequired);
+        }
+        await this._initPromise;
+        this._cruxDomain = await this._cruxDomainRepository.create(domain, this._configKeyManager);
+        return;
+    }
+
+    public getCruxDomainState = async (): Promise<DomainRegistrationStatus> => {
+        await this._initPromise;
+        if (!this._cruxDomain) {
+            throw errors.ErrorHelper.getPackageError(null, errors.PackageErrorCode.MissingCruxDomain);
+        }
+        return this._cruxDomain.status;
+    }
+
+    public getNameServiceConfig = async () => {
+        await this._initPromise;
+        if (!this._cruxDomain) {
+            throw errors.ErrorHelper.getPackageError(null, errors.PackageErrorCode.MissingCruxDomain);
+        }
+        return this._cruxDomain.nameServiceConfig;
+    }
+
+    public getAssetMap = async (): Promise<IClientAssetMapping> => {
+        await this._initPromise;
+        if (!this._cruxDomain) {
+            throw errors.ErrorHelper.getPackageError(null, errors.PackageErrorCode.MissingCruxDomain);
+        }
+        return this._cruxDomain.assetMap;
+    }
+
+    public getGlobalAssetList = async (): Promise<configurationService.IGlobalAssetList> => {
```

Comment:
As discussed, we have two `globalAssetList`. This method should be converted to
`getReferenceGlobalAssetList`.

Reply:
I have exported the CruxSpec class. Amit can use the reference list from there, instead of that being a method in the client.